tracing/opentracing/grpc.go

```diff
@@ -0,0 +1,56 @@
+package opentracing
+
+import (
+	"github.com/opentracing/opentracing-go"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/metadata"
+)
+
+// ToGRPCRequest returns a grpc RequestFunc that injects an OpenTracing Span
+// found in `ctx` into the grpc Metadata. If no such Span can be found, the
+// RequestFunc is a noop.
+func ToGRPCRequest(tracer opentracing.Tracer) func(ctx context.Context, md *metadata.MD) context.Context {
+	return func(ctx context.Context, md *metadata.MD) context.Context {
+		if span := opentracing.SpanFromContext(ctx); span != nil {
+			// There's nothing we can do with an error here.
+			_ = tracer.Inject(span, opentracing.TextMap, metadataReaderWriter{md})
+		}
+		return ctx
+	}
+}
```

Comment:
@peterbourgon I think a cross-tracing-system wire-format spec is difficult to achieve for fundamental reasons... That said, I am pleased with the approach in OpenTracing where we standardize on the concepts and basic formats (i.e., text maps and binary blobs) for propagation but leave the actual field choices to the tracing implementations... it provides interop at the instrumentation layer without forcing tracing systems to use precisely the same data model (because they don't!).